src/main/java/org/isf/medicalstock/manager/MovBrowserManager.java

```diff
@@ -122,4 +152,53 @@ private void check(LocalDateTime from, LocalDateTime to, String errMsgKey) throw
 			}
 		}
 	}
+
+	/**
+	 * Get the last Movement.
+	 *
+	 * @return the retrieved movement.
+	 * @throws OHServiceException
+	 */
+	public Movement getLastMovement() throws OHServiceException {
+		return ioOperations.getLastMovement();
+	}
+
+	/**
+	 * Deletes the last Movement.
+	 *
+	 * @param lastMovement - the last movement to delete
+	 * @throws OHServiceException
+	 */
+	@Transactional
+	public void deleteLastMovement(Movement lastMovement) throws OHServiceException {
+
+		MovementType movType = medicalDsrStockMovTypeManager.getMovementType(lastMovement.getType().getCode());
+		Lot lot = lastMovement.getLot();
+		Medical medical = lastMovement.getMedical();
+		int medicalCode = medical.getCode();
+		int quantity = lastMovement.getQuantity();
+		if (movType.getType().contains("+")) {
+			medical.setInqty(medical.getInqty() - quantity);
+			medicalsIoOperation.updateMedical(medical);
+			List<Movement> movementWithSameLot = ioOperations.getMovementByLot(lot);
+			if (movementWithSameLot.size() == 1) {
+				lotRepository.deleteById(lot.getCode());
+			}
+			ioOperations.deleteMovement(lastMovement);
+		} else {
+			Ward ward = lastMovement.getWard();
+			String wardCode = ward.getCode();
+			MedicalWard medWard = movWardBrowserManager.getMedicalWardByWardAndMedical(wardCode, medicalCode, lot.getCode());
+			medWard.setIn_quantity(medWard.getIn_quantity() - quantity);
+			if (medWard.getIn_quantity() == 0 && medWard.getOut_quantity() == 0) {
+				movWardBrowserManager.deleteMedicalWard(medWard);
+			} else {
+				movWardBrowserManager.updateMedicalWard(medWard);
+			}
+			medical.setOutqty(medical.getOutqty() - quantity);
+			medicalsIoOperation.updateMedical(medical);
```

Comment:
Here we have to care also about `oh_medicaldsrstockmovward` that we totally forget. 

**To test the new feature please follow these steps:**

1. start from a fresh demo DB
2. add a number of charges and discharges in "Pharmaceuticals Stock"
3. delete all of them in reverse order (you should be obliged to do so)
4. export a new dump and compare it with `load_demo_data.sql` script

At step 4, only `*_LOCK`, `*_LAST_MODIFIED_DATE`, `*_LAST_MODIFIED_BY` and `oh_session_audit` table should differ; quantities and number of `LOT`s, `MEDICAL`s and `MEDICALWARD`s as well as `MOVEMENT`s and `MOVEMENTWARD`s should be unchanged.

**To test _denied_ deletions:**

1. start from a fresh demo DB
2. add a number of charges and discharges in "Pharmaceutical Stock"
3. add a number of discharges (either to Patients, Internal or Wards) in "Pharmaceuticals Stock Ward"
4. deletion should be **denied** and we will need also OP-1169 to restore the initial dump

Reply:
hello @mwithi  I followed all the steps above and everything works fine.

Reply:
hello @mwithi it is okay now please can you test it again if it is okay.

Reply:
where is the part caring about `oh_medicaldsrstockmovward` as I was suggesting?

Reply:
Hello 
[@Alessandro Domanico](https://openhospitalworkspace.slack.com/team/US2C2NPA9)
 after investigation of you suggestion I have made a remark that in the Income tab In pharmaceutical stock ward,  the values that we have  is that tab came from Movement not from MovementWard. When we do now the new movement in pharmaceutical stock ward we save now that in Movement Ward table. for exemple when we discharge something to central shop to a ward the MovementWard is not create so we can not delete the movement in MovementWard when the last movement is discharge. see the following picture. 
![movWard](https://github.com/informatici/openhospital-core/assets/101590821/95a137a7-aeaa-48e9-a3fc-62ef05511458)

Reply:
you're right, the Income Tab shows the movement from main pharmacy to the specific ward, while the `oh_medicaldsrstockmovward` table is used only for the ward outgoing movements

we have to block the deletion of a discharge movement from main pharmacy to a ward if the same medical has been furtherly discharged to a patient/ward/internal